app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/mini_game/PlasmaLeakActivity.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    /**
+     * What happens if the lab radio button is pressed.
+     *
+     * @param view, the view of the page.
+     */
+    public void switchLab(View view) {
+            labOn = true;
+            deckOn = false;
+            armoryOn = false;
+            engineOn = false;
+    }
+
+    /**
+     * What happens if the armory radio button is pressed.
+     *
+     * @param view, the view of the page.
+     */
+    public void switchArmory(View view) {
+            labOn = false;
+            deckOn = false;
+            armoryOn = true;
+            engineOn = false;
+    }
+
+    /**
+     * What happens if the deck radio button is pressed.
+     *
+     * @param view, the view of the page.
+     */
+    public void switchDeck(View view) {
+            labOn = false;
+            deckOn = true;
+            armoryOn = false;
+            engineOn = false;
+    }
+
+    /**
+     * What happens if the engine radio button is pressed.
+     *
+     * @param view, the view of the page.
+     */
+    public void switchEngine(View view) {
+            labOn = false;
+            deckOn = false;
+            armoryOn = false;
+            engineOn = true;
+    }
+
+    /**
+     * Step 1 of the mini-game process.
+     *
+     * @param view, the view of the page.
+     */
+    public void removeBolts(View view) {
+        if(progress.getProgress() < 25) {
+            if(removedbolts != 5 ) {
+                progress.incrementProgressBy(5);
+                removedbolts++;
+            }
+        }
+    }
+
+    /**
+     * Step 2 of the mini-game process.
+     *
+     * @param view, the view of the page.
+     */
+    public void repairPipe(View view) {
+        if(progress.getProgress() < 75 && progress.getProgress() >=25) {
+            if(removedbolts == 5) {
+                progress.incrementProgressBy(5);
+            }
+        }
+    }
+
+    /**
+     * Step 3 of the mini-game process.
+     *
+     * @param view, the view of the page.
+     */
+    public void insertBolts(View view) {
+        if(progress.getProgress() < 100 && progress.getProgress() >=75) {
+            if(removedbolts != 0 ) {
+                progress.incrementProgressBy(5);
+                removedbolts--;
+            }
+        }
+    }
+
+    /**
+     * Sends a faulty message.
+     *
+     */
+    public void sendFailure() {
+        game.sendMessage(new EventCompletedMessage(-1));
+        Intent intent = new Intent(this, LocationEngineroomActivity.class);
+        startActivity(intent);
+    }
+
+
+    /**
+     * If the event is completed, send the message to the server.
+     *
+     * @param view, the view of the page.
+     */
+    public void complete(View view) {
+        if(progress.getProgress() == 100) {
+            EventCompletedMessage message = new EventCompletedMessage(EventTypes.PLASMA_LEAK.ordinal());
+            Map<String, Integer> parameters = new HashMap<String, Integer>();
+            if(labOn) {
```

Comment:
Why al the booleans and then converting them to integers? Can't you just store the integer directly? For example if you press switchLab you just set the sector int attribute to 0.